lib_test/test_httpaf_server.ml

```diff
@@ -113,9 +113,21 @@ let streaming_response =
 
 ;;
 
+let malformed_requests_tests = [
+    "single GET, malformed request"
+  , `Quick
+  , Simulator.test_server
+      ~handler: (basic_handler "")
+      ~input:   [ `Raw [ "GET / HTTP/1.1\r\nconnection: close\r\nX-Other-Header : shouldnt_have_space_before_colon\r\n\r\n" ]
```

Comment:
I expected that this test case would pass with the current code by simply propagating the errors from the parser to `Server_connection` ([diff](https://github.com/inhabitedtype/httpaf/pull/98/files#diff-b67911d869b868862edd1579e631056bR319)). However, the test reveals that the connection is closed before the error response is written.